### PR TITLE
Bump chisel-circt to 0.8.0 (was 0.5.0)

### PR DESCRIPTION
**Pull request by mwachs5**
### Contributor Checklist

- [ ] Did you add Scaladoc to every public function/method?
- [ ] Did you add at least one test demonstrating the PR?
- [ ] Did you delete any extraneous printlns/debugging code?
- [ ] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [ ] Did you state the API impact?
- [ ] Did you specify the code generation impact?
- [ ] Did you request a desired merge strategy?
- [ ] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement

   - performance improvement     

#### API Impact

There were some breaking changes in this bump. Here are the corresponding chisel-circt release notes:

[v0.8.0: v0.7.1](https://github.com/sifive/chisel-circt/releases/tag/v0.8.0)
@[seldridge](https://github.com/seldridge) seldridge tagged this 2 days ago
- Fix a bug in NoDedupAnnotation handling
- Remove CIRCT Handover Functionality

[v0.7.0: Version 0.7.0](https://github.com/sifive/chisel-circt/releases/tag/v0.7.0)
@[seldridge](https://github.com/seldridge) seldridge tagged this 25 days ago
- Bump Chisel to 3.5.5
- Bump CIRCT to 1.22.1
- Add support for Chisel's Trace API
- Improve APIs for passing options to circt.stage.ChiselStage

[v0.6.0: Bump to CIRCT 1.15.0](https://github.com/sifive/chisel-circt/releases/tag/v0.6.0)
@[youngar](https://github.com/youngar) youngar tagged this Sep 8
The command line options have been updated for this version of CIRCT.

#### Backend Code Generation Impact

It changes if you were using chisel-circt, most like for small test cases and/or docs

#### Desired Merge Strategy

 - Squash: The PR will be squashed and merged (choose this if you have no preference. 

#### Release Notes

Bump chisel-circt from 0.5.0 to 0.8.0. No further upstream development on chisel-circt is planned.

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (Bug fix: `3.4.x`, [small] API extension: `3.5.x`, API modification or big change: `3.6.0`)?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you do one of the following when ready to merge:
  - [ ] Squash: You/ the contributor `Enable auto-merge (squash)`, clean up the commit message, and label with `Please Merge`.
  - [ ] Merge: Ensure that contributor has cleaned up their commit history, then merge with `Create a merge commit`.

### CODE REVIEW

**Comment by mwachs5** [comment]
by the way, even though this was brought in with `git subtree` originally things moved around enough that I just did the actual diff-applying via the VSCode Diff Folders Plugin: https://marketplace.visualstudio.com/items?itemName=L13RARY.l13-diff

**Comment by mwachs5** [comment]
TBD if we actually want to backport to 3.5.x given that it bumps the circt version

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
         jvm: ["8"]
         scala: ["2.13.10", "2.12.17"]
         espresso: ["2.4"]
-        circt: ["sifive/1/21/0"]
```
Comment (seldridge) [inline .github/workflows/test.yml:22]: Does 1/24/0 work?

**Review by seldridge — APPROVED** [review]
I just took a glance at this. As long as this passes tests, it should be good to go!

Thanks for keep this coherent. 💯

File: .github/workflows/test.yml (hunk 1 above)
Comment (mwachs5) [reply .github/workflows/test.yml:22]: probably, but we can do that in a follow-up to keep this tidy?

**Pull request merged**